src/Gui/CommandMacro.cpp

```diff
@@ -54,13 +56,27 @@ StdCmdDlgMacroRecord::StdCmdDlgMacroRecord()
 void StdCmdDlgMacroRecord::activated(int iMsg)
 {
     Q_UNUSED(iMsg);
-    Gui::Dialog::DlgMacroRecordImp cDlg(getMainWindow());
-    cDlg.exec();
+    if (!getGuiApplication()->macroManager()->isOpen()){
+        Gui::Dialog::DlgMacroRecordImp cDlg(getMainWindow());
+        if (cDlg.exec() && getAction()) {
+            getAction()->setIcon(Gui::BitmapFactory().iconFromTheme("media-playback-stop"));
+            getAction()->setText(QString::fromLatin1(QT_TR_NOOP("S&top macro recording")));

```

Comment:
I don't think using the NOOP version is correct here, I think you need the actual translated text (so using `tr()` instead).

Reply:
tr() is a QObject function. So it would be QObject::tr(...). Is that correct ? I pushed a commit. If that's ok let me know and I'll squash.

Reply:
If you directly call the QObject version doesn't it get the context wrong? I thought this class was derived from QObject (sorry, not at a computer right now to check).

Reply:
I don't know. Sorry I don't know the translation code at all so it is not clear when to use one or another. Though what is sure is that Gui::Command are not QObjects

Reply:
Ah, sorry, I say the "Dlg" and immediately thought of the Dialog classes. So using QObject::tr() should be correct and put it in the right context. I'll make a note to check on the next translation cycle.

Reply:
I have done so, squashed and and force pushed. Please have a look thanks.